File: session_py/examples/arrow_example.py
import sys

sys.path.insert(0, "src")

from session_py import Arrow, Line


def main():
    line = Line(0.0, 0.0, 0.0, 10.0, 0.0, 10.0)
    arrow = Arrow(line, 1.0)

    print("=== Cylinder/Pipe Generation Example ===\n")

    v_vertices, v_faces = arrow.mesh.to_vertices_and_faces()

    for vertex in v_vertices:
        print(f"{vertex.x} {vertex.y} {vertex.z}")

    print("Faces:")
    for face in v_faces:
        print(f"{face[0]} {face[1]} {face[2]}")


if __name__ == "__main__":
    main()

File: session_rust/examples/arrow_example.rs
use session_rust::{Arrow, Line};

fn main() {
    let line = Line::new(0.0, 0.0, 0.0, 10.0, 0.0, 10.0);
    let arrow = Arrow::new(line, 1.0);

    println!("=== Cylinder/Pipe Generation Example ===\n");

    let (v_vertices, v_faces) = arrow.mesh.to_vertices_and_faces();

    for vertex in &v_vertices {
        println!("{} {} {}", vertex.x(), vertex.y(), vertex.z());
    }

    println!("Faces:");
    for face in &v_faces {
        println!("{} {} {}", face[0], face[1], face[2]);
    }
}
